tool/tctl/common/app_command.go

```diff
@@ -63,11 +74,44 @@ func (c *AppsCommand) TryRun(cmd string, client auth.ClientI) (match bool, err e
 
 // ListApps prints the list of applications that have recently sent heartbeats
 // to the cluster.
-func (c *AppsCommand) ListApps(client auth.ClientI) error {
-	servers, err := client.GetApplicationServers(context.TODO(), apidefaults.Namespace)
-	if err != nil {
+func (c *AppsCommand) ListApps(clt auth.ClientI) error {
+	ctx := context.TODO()
+
+	var labels map[string]string
+	var err error
+	if c.labels != "" {
```

Comment:
Wouldn't `ParseLabelSpec` work fine on an empty string?

Reply:
you're right, i removed the checks